lib/client.js

```diff
@@ -710,10 +721,12 @@ Request.prototype._end = function() {
   // progress
   const handleProgress = (direction, e) => {
 
-    clearTimeout(self._uploadTimeoutTimer);
     if (e.total > 0) {
       e.percent = e.loaded / e.total * 100;
+    } else if(e.total = 100){
```

Comment:
You must write unit tests for it, because you've written non-working code second time in a row. 

This is not a comparison, but an assignment (the condition is always true in this case).

Total is not a percentage, but amount of data to be transferred, so a comparison with a 100 byte file size doesn't make any sense.

If you're only clearing the timer without resetting, it doesn't add much value beyond response timeout. You could just add upload time to response timeout.

Reply:
yep you right, honestly I guess we could add some linter in the project to avoid this mistake. What do you think?

Reply:
I change it but is actually not clear to me the part :  

If you're only clearing the timer without resetting, it doesn't add much value beyond response timeout. You could just add upload time to response timeout.

Do you have suggestions for the test?

Reply:
If you set timer once, and clear it once, then it's a one fixed length of time from start to finish. It doesn't change the timeout if there's one progress event or 100 progress events in the meantime.

The response timeout timer has the same inflexible behavior - it's set once, and cleared once. The only difference is that it's cleared in readystate rather than progress event, but that's not a major difference most of the time. So effectively your approach doesn't bring anything new.

The adaptive approach would clear timer and set a new one after every single progress event. (that's why I've suggested using a bound method, btw). That would make it adaptive to rate of events.

Reply:
For the "lint", we use unit tests as the lint. See existing tests. Write an endpoint for our built-in server, and then write code for success and failure case you expect.

Reply:
the lint would be different, ti will promote a common style across all the files http://jshint.com/docs/ . Tests are great but they are not style related they only guarantee that the functionality respect the wanted behaviour. I you see in your codebase there are different case where you use somewhere single/double quote ==/=== and other similars. Have a linter will guarantee a common style across all the contributors. Another usefull file for the indentation could be have an .editorconfig file where is described where any kind of file is formatted

Reply:
@kornelski your idea is to add an adaptive timeout similar the TCP one? http://www.pcvr.nl/tcpip/tcp_time.htm#21_0 if yes we should understand which strategy apply